git_test.go

```diff
@@ -4,72 +4,198 @@ import (
 	"context"
 	"io"
 	"net/http/httptest"
+	"net/url"
 	"os"
 	"testing"
 	"time"
 
 	"github.com/coder/envbuilder"
 	"github.com/coder/envbuilder/gittest"
+	"github.com/go-git/go-billy/v5"
 	"github.com/go-git/go-billy/v5/memfs"
 	"github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/plumbing/object"
+	githttp "github.com/go-git/go-git/v5/plumbing/transport/http"
 	"github.com/stretchr/testify/require"
 )
 
 func TestCloneRepo(t *testing.T) {
 	t.Parallel()
 
-	t.Run("Clones", func(t *testing.T) {
-		t.Parallel()
+	for _, tc := range []struct {
```

Comment:
review: trying to cover all cases outlined here: 

https://github.com/coder/envbuilder?tab=readme-ov-file#git-authentication